### PR TITLE
Synapse workspace to sync with GA version

### PR DESCRIPTION
**Pull request by krutikasheth1029**

### CODE REVIEW

**Comment by coolswatish** [comment]
Please validate the workspace deployment using this URL once: https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fkrutikasheth1029%2FSynapse%2Fpatch-3%2FManage%2FDeployWorkspace%2Fworkspace%2Fazuredeploy.json

Also, I see that some parameters are pre-filled incorrectly. Can you please check them as well?
image

**Comment by cool-mist** [comment]
- Remove `virtualNetworkProfile`. It is not being used and will result in a bad request if customer sets it to non empty string by mistake.
- Add 'publicNetworkAccess' property.
- Add `defaultDataLakeStorage.resourceId` property.
- Add `defaultDataLakeStorage.createManagedPrivateEndpoint` property.
- Can the encryption json be expanded ([as per this](https://docs.microsoft.com/en-us/rest/api/synapse/workspaces/create-or-update#encryptiondetails)) for ease of use.

**Comment by krutikasheth1029** [comment]
@cool-mist Hi Surya, I have made the changes as per your comments and also discussed with Swatish.

- **virtualNetworkProfile** has been removed
- resourceId and createManagedPrivateEndpoint have been added, but we will not be using Encryption, creating managed Vnet  endpoints functionalities for our product. These will be set to false by default

**Comment by cool-mist** [comment]
Add `publicNetworkAccess` and `managedVirtualNetworkSettings`

**Review by cool-mist — APPROVED** [review]

File: Manage/DeployWorkspace/workspace/azuredeploy.json [hunk 1]
```diff
@@ -55,6 +55,34 @@
...
+        },
+        "storageRoleUniqueId": {
+            "type": "string",
+            "defaultValue": "[newGuid()]"
+        },
+        "allowAllConnections": {
+            "type": "bool",
+            "defaultValue": true
+        },
+        "defaultAdlsGen2AccountResourceId": {
```
Comment (jovanpop-msft) [inline Manage/DeployWorkspace/workspace/azuredeploy.json:75]: We should not ask users to search some resource id in the quickstart

File: Manage/DeployWorkspace/workspace/azuredeploy.json [hunk 2]
```diff
@@ -55,6 +55,34 @@
...
+        },
+        "managedResourceGroupName": {
+            "type": "string",
+            "defaultValue": ""
+        },
+        "setSbdcRbacOnStorageAccount": {
+            "type": "bool",
+            "defaultValue": false
+        },
+        "storageRoleUniqueId": {
```
Comment (jovanpop-msft) [inline Manage/DeployWorkspace/workspace/azuredeploy.json:67]: If this is autogenerated it should be an internal variable

File: Manage/DeployWorkspace/workspace/azuredeploy.json [hunk 3]
```diff
@@ -55,6 +55,34 @@
             "metadata": {
                 "description": "The encryption object containing your customer-managed key used for double encryption (optional)."
             }
+        },
+        "managedResourceGroupName": {
```
Comment (jovanpop-msft) [inline Manage/DeployWorkspace/workspace/azuredeploy.json:59]: in the quickstart this is always resyrce().location

**Pull request merged**